pennylane/pauli/utils.py

```diff
@@ -81,14 +80,22 @@ def is_pauli_word(observable):
     >>> is_pauli_word(qml.PauliZ(0) @ qml.Hadamard(1))
     False
     """
-
     if not isinstance(observable, Observable):
         return False
 
     pauli_word_names = ["Identity", "PauliX", "PauliY", "PauliZ"]
     if isinstance(observable, Tensor):
         return set(observable.name).issubset(pauli_word_names)
 
+    if isinstance(observable, Hamiltonian):
+        terms_pauli_word = []
+        for _, ob in zip(*observable.terms()):
+            if isinstance(ob, Tensor):
+                terms_pauli_word.append(set(ob.name).issubset(pauli_word_names))
+            else:
+                terms_pauli_word.append(ob.name in pauli_word_names)
```

Comment:
I believe can use recursion here:
```suggestion
            terms_pauli_word.append(is_pauli_word(ob))
```

Reply:
Nice catch!